libs/core/src/lib/message-box/message-box-semantic-icon/message-box-semantic-icon.component.ts

```diff
@@ -0,0 +1,64 @@
+import { ChangeDetectionStrategy, Component, Input, Optional } from '@angular/core';
+import {
+    MESSAGE_BOX_CONFIGURABLE_ELEMENT,
+    MessageBoxConfig,
+    MessageBoxConfigurableElement
+} from '../utils/message-box-config.class';
+
+/**
+ * Message box element representing the semantic icon in the message box header.
+ *
+ * ```html
+ * <fd-message-box-semantic-icon></fd-message-box-semantic-icon>
+ * ```
+ */
+@Component({
+    selector: 'fd-message-box-semantic-icon',
+    template: `<i [ngClass]="'sap-icon--' + _semanticIcon" role="presentation"></i>`,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [
+        { provide: MESSAGE_BOX_CONFIGURABLE_ELEMENT, useExisting: MessageBoxSemanticIconComponent, multi: true }
+    ]
+})
+export class MessageBoxSemanticIconComponent implements MessageBoxConfigurableElement {
+    /** Custom semantic icon */
+    @Input()
+    glyph: string;
+
+    /** @hidden */
+    constructor(@Optional() public messageBoxConfig: MessageBoxConfig) {
+        this.messageBoxConfig = this.messageBoxConfig || {};
+    }
+
+    /** @hidden */
+    get _showSemanticIcon(): boolean {
+        return this.messageBoxConfig.showSemanticIcon;
+    }
+
+    /** @hidden */
+    get _semanticIcon(): string {
```

Comment:
Minor suggestion, but I think in this form it's more readable:
```
get _getIcon(): string {
    return this.glyph || this.messageBoxConfig.customSemanticIcon || this._semanticIcon;
}

get _semanticIcon(): string {
    switch (this.messageBoxConfig.type) {
       case 'error':
           return 'message-error';
       case 'success':
           return 'message-success';
       case 'warning':
           return 'message-warning';
       case 'information':
           return 'message-information';
       case 'confirmation':
           return 'question-mark';
       default:
           return '';
       }
}
```
Original form is also fine, but it looks like we don't need default value in the switch statement.